### DISCUSSION TITLE
Helm upgrade finishes before deployment to all pods has finished.

### DISCUSSION THREAD

**Discussion by ddanda** [openkruise/kruise#1310] [General]
I've noticed that the helm upgrade command with Advanced Statefulsets returns quickly. The upgrade is complete in 97 seconds on my cluster while releasing new images. Previously the install\upgrade command used to wait until the deployment is completed to every pod. Currently with advanced statefulsets, helm returns completed status while the deployment is in progress on one of the pods. 

Has anyone seen this behaviour ?

**Comment by furykerry** [comment]
did you use the --wait arg ? 

      --wait                                       if set, will wait until all Pods, PVCs, Services, and minimum number of Pods of a Deployment, StatefulSet, or ReplicaSet are in a ready state before marking the release as successful. It will wait for as long as --timeout

**Comment by zmberg** [comment]
@ddanda Helm currently don't support to wait for custom resources ready, like Advanced StatefulSet, and you can add [post-hook](https://helm.sh/docs/topics/charts_hooks/) to check whether Advanced StatefulSet is ready.
